python/tvm/autotvm/measure/measure_methods.py

```diff
@@ -242,6 +243,22 @@ def __init__(
 
         self.executor = LocalExecutor(timeout=timeout * (self.n_parallel + 1))
 
+    @property
+    def ref_input(self):
+        """Fixed input for tuning special operators."""
+        return self._ref_input if hasattr(self, "_ref_input") else None
```

Comment:
seems as though we could just set `self._ref_input` to `None` in `__init__` if it is not set. also, why this style of setting, rather than taking as an `__init__` arg?

Reply:
The aim for using `property` is to raise warning whenever setting the attribute. In prior versions, there is no entry for `ref_input` in the ctor arg, and it will be set after the object is initialized. Thus IMO it is reasonable to do the warnings and check we want in a separate setter function rather than the ctor. But yes, we can add an optional entry in the ctor, and the `if` expression can definitely be resolved.

Reply:
ah that's right. i think it would be a better design to keep with the pattern of ctor arg and not modify attributes of the runner after construction. however, if the value needs to change during tuning, I'm fine with the setter--just would suggest to drop `hasattr` here and instead, in ctor, include: `self._ref_input = None`

Reply:
just come across a new idea. it would be more reasonable to set `ref_input` on `measure_option` rather than `RPCRunner`; what's more, it can be implemented by just forwarding `ref_input` to the runner object, and we can still make the warning when setting the runner object. `Runner(ref_input=*)` seems somewhat strange to me.

Reply:
that makes sense to me @Tantalus13A98B5F ! i think you could also potentially put it on `MeasureInput` if you wanted to encapsulate the interface further.